### ISSUE TITLE
Key Error: u'Constant339'

### ISSUE THREAD

**Issue by joninsky** [onnx/onnx-coreml#374]
I'm trying to convert the [emotion_ferplus.onnx model](https://github.com/onnx/models/tree/master/emotion_ferplus) into CoreML.

I installed `onnx-coreml` using pip. I then downloaded the model and put it into a folder with this python script:

```python
from onnx_coreml import convert
import onnx
modelFile = onnx.load('emotion_ferplus.onnx')
mlmodel = convert(modelFile)
mlmodel.save('emotion_ferplus.mlmodel')
```

I run the python script but get the error logs:

```shell
Traceback (most recent call last):
  File "script_onnx.py", line 4, in <module>
    mlmodel = convert(modelFile)
  File "/Users/Nomad/Library/Python/2.7/lib/python/site-packages/onnx_coreml/converter.py", line 458, in convert
    _convert_node(builder, node, graph, err)
  File "/Users/Nomad/Library/Python/2.7/lib/python/site-packages/onnx_coreml/_operators.py", line 1755, in _convert_node
    return converter_fn(builder, node, graph, err)
  File "/Users/Nomad/Library/Python/2.7/lib/python/site-packages/onnx_coreml/_operators.py", line 246, in _convert_sub
    _convert_broadcast_op(builder, node, graph, err, "ADD")
  File "/Users/Nomad/Library/Python/2.7/lib/python/site-packages/onnx_coreml/_operators.py", line 68, in _convert_broadcast_op
    ranks = [len(graph.onnx_coreml_shape_mapping[input_]) for input_ in node.inputs]
KeyError: u'Constant339'
```

I'm an iOS developer who is new the the world of machine learning so I'm probably missing something obvious. Am I supposed to know what the inputs for the model are and pass those in as arguments?

Thanks for the help.

**Comment by joninsky** [comment]
`Dockerfile` that I was able to reproduce the issue in.

```ruby
FROM python:2.7

RUN pip install -U onnx-coreml

COPY ./emotion_ferplus.onnx ./

COPY ./convert.py ./

RUN ["python", "convert.py"]
```

The `convert.py` file has the python code that is in the original question.

**Comment by dianxin556** [comment]
I meet the same problem ,and i find the lost key is in input tensor not in node.inputs

**Comment by Ciyou** [comment]
Same problem here. Does anyone have a solution?

**Comment by Ciyou** [comment]
`File "/Users/***/Library/Python/3.6/lib/python/site-packages/onnx_coreml/_operators.py", line 741, in _convert_bn
    mapp = graph.onnx_coreml_shape_mapping[node.inputs[0]]
KeyError: '244'`

What do these keys mean in `graph.onnx_coreml_shape_mapping`?

**Comment by Ciyou** [comment]
Now I figured out that the keys are `node ID` in a model. And my problem occurred in a BatchNormalize node where get its input from a Conv node which has the node id `244`. But why there's no `244` in the shape map?

**Comment by yehorchankov** [comment]
> Now I figured out that the keys are `node ID` in a model. And my problem occurred in a BatchNormalize node which gets its input from a Conv node which has the node id `244`. But why there's no `244` in the shape map?

I have the same issue with the node of type Slice as reported by @Ciyou. In my case: `KeyError: '367'`. 
I used [netron](https://github.com/lutzroeder/netron) to visualize my onnx graph and I found the node is presented in the graph, but its ID is not presented in `graph.shape_dict` keys. 
Is there any solution or workaround to the issue?

**Comment by yehorchankov** [comment]
While my issue seems unrelated to the original question, I have fixed the problem in my situation. See "Fixes" here #458 (KeyError during Converting Node Type Slice).

**Comment by bhushan23** [comment]
Issue not producible  
@joninsky feel free to re-open if you still face issues
